### DISCUSSION TITLE
Replace with Register in nvim

### DISCUSSION THREAD

**Discussion by whatwouldmarvindo** [LazyVim/LazyVim#1371] [Q&A]
Hello fellow LazyVimmers,

I am in transition from VSCode to lazyvim and I am having trouble archieving the same behaviour when I comes to replacing test with a register (mostly unnamed register). Before I was making use of the fairly popular vimscript [replaceWithRegister](https://github.com/vim-scripts/ReplaceWithRegister) but I cannot figure out how to implement it in my config? Do I have save the source code manually? I guess I need the vimcmd() function somewhere but really need some guidance.

I would love to extend the recipes page on the website for anyone having a similar problem...

**Comment by dpetka2001** [comment]
While in insert mode you can press `<C-r>` and it will show you the contents of the registers. You just then type the letter of the register you want. I'm not familiar with the plugin you mentioned.

**Reply by whatwouldmarvindo** [reply]
This is a really nice hint! Thank you very much for your answer.

The plugin I referenced replaces text with a register. So for example if I want to replace a word I would do `griw`
(`gr` = goReplace)
also works with referencing a register before the command. The workaround for this at the moment would be `viwp` and I am looking for something similar like before

**Reply by dpetka2001** [reply]
`gr` is by default bound to `lsp_references` whenever a LSP server is attached. So you might want to change that to another keybinding. Something like
```lua
return {
  {
    "vim-scripts/ReplaceWithRegister",
    keys = {
      { "<leader>gr", "<Plug>ReplaceWithRegisterOperator", desc = "ReplaceWithRegisterOperator" },
    },
  },
}
```
Change the keybinding to whatever you want, just make sure there is no conflict with other default Lazyvim keybindings.

**Reply by whatwouldmarvindo** [reply]
Thank you so much! That really helps!

With a strange coincidence I just found this plugin called [mini.operators](https://github.com/echasnovski/mini.nvim/blob/main/readmes/mini-operators.md). I was able to set it up and change the key binding so it doesn't conflict with `lsp_references`

**Reply by MinmoTech** [reply]
As an alternative solution, the keybind `gr` for `lsp_references` can be disabled with the following config:
```lua
return {
  { "vim-scripts/ReplaceWithRegister" },
  -- LSP keymaps
  {
    "neovim/nvim-lspconfig",
    init = function()
      local keys = require("lazyvim.plugins.lsp.keymaps").get()
      keys[#keys + 1] = { "gr", false }
    end,
  },
}
```

**Reply by con-f-use** [reply]
> ```lua
> return {
>   { "vim-scripts/ReplaceWithRegister" },
>   -- LSP keymaps
>   {
>     "neovim/nvim-lspconfig",
>     init = function()
>       local keys = require("lazyvim.plugins.lsp.keymaps").get()
>       keys[#keys + 1] = { "gr", false }
>     end,
>   },
> }
> ```

This just spits a bunch of errors for me:

![image](https://github.com/LazyVim/LazyVim/assets/11145016/5fae590f-d014-48fa-89cb-3957b341302a)

**Comment by huo-feng-ding** [comment]
{
		'inkarkat/vim-ReplaceWithRegister',
		config = function()
			vim.api.nvim_set_keymap('n', '<Leader>gr', '"+gr', { noremap = false, silent = true })
		end,
	},

**Comment by cepehang** [comment]
I managed to replace `gr` with the replace operator of `mini.operators`. The only thing I did not get to work was getting `which-key` to show `+Replace operator` as an intermediate keybinding on the *first* press of `gr`. It shows the description fine after the first replacement : 

```lua
return {
  {
    "neovim/nvim-lspconfig",
    opts = function()
      local keys = require("lazyvim.plugins.lsp.keymaps").get()
      keys[#keys + 1] = { "gr", false }
    end,
  },
  { "echasnovski/mini.operators", keys = { { "gr", desc = "Replace operator" } }, opts = {} },
}
```
